model/settings/service.go

```diff
@@ -2,53 +2,115 @@ package settings
 
 import (
 	"fmt"
+	"net/url"
+	"strings"
+	"time"
 
-	"github.com/cozy/cozy-stack/pkg/consts"
+	"github.com/cozy/cozy-stack/model/instance"
+	"github.com/cozy/cozy-stack/model/token"
 	"github.com/cozy/cozy-stack/pkg/couchdb"
+	"github.com/cozy/cozy-stack/pkg/emailer"
 	"github.com/cozy/cozy-stack/pkg/prefixer"
 )
 
+const TokenExpiration = 7 * 24 * time.Hour
+
 var (
 	ErrInvalidType = fmt.Errorf("invalid type")
 	ErrInvalidID   = fmt.Errorf("invalid id")
 )
 
+// Storage used to persiste and fetch settings data.
+type Storage interface {
+	setInstanceSettings(inst prefixer.Prefixer, doc *couchdb.JSONDoc) error
+	getInstanceSettings(inst prefixer.Prefixer) (*couchdb.JSONDoc, error)
+}
+
 // SettingsService handle the business logic around "settings".
 //
 // This service handle 2 structured documents present in [consts.Settings]
 // - The "instance settings" ([consts.InstanceSettingsID])
 // - The "bitwarden settings" ([consts.BitwardenSettingsID]) (#TODO)
 type SettingsService struct {
+	emailer  emailer.Emailer
+	instance instance.Service
+	token    token.Service
+	storage  Storage
 }
 
 // NewService instantiates a new [SettingsService].
-func NewService() *SettingsService {
-	return &SettingsService{}
+func NewService(
+	emailer emailer.Emailer,
+	instance instance.Service,
+	token token.Service,
+	storage Storage,
+) *SettingsService {
+	return &SettingsService{emailer, instance, token, storage}
 }
 
-func (s *SettingsService) GetInstanceSettings(inst prefixer.Prefixer) (*couchdb.JSONDoc, error) {
-	doc := &couchdb.JSONDoc{}
-
-	err := couchdb.GetDoc(inst, consts.Settings, consts.InstanceSettingsID, doc)
+// PublicName returns the settings' public name or a default one if missing
+func (s *SettingsService) PublicName(db prefixer.Prefixer) (string, error) {
+	doc, err := s.storage.getInstanceSettings(db)
```

Comment:
While we are doing refactoring, I think it would be useful to have a way to get the public name and the email while fetching only once the settings document from CouchDB.

For example, https://github.com/cozy/cozy-stack/blob/master/model/sharing/sharing.go#L164-L171 is probably making two CouchDB requests.

Reply:
Totally agree! This is completely on my bucket list but it was not a requirement for this PR and I tried to stay as focus as possible and as I see it this is not a trivial change. 

My vision for this issue would be to modfify the method `GetInstanceSettings` and make it return a structured `InstanceSettings` struct with the method `PublicName(db prefixer.Prefixer)` attached to it.